test/common/change_notifier_test_util.dart

```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/foundation.dart';
+
+extension ChangeNotifierTextExtensions on ChangeNotifier {
+  void verifyStateInOrder(
```

Comment:
spoko opcja, jak działa dokładnie ten listener w środku?

Reply:
generalnie przy każdym kolejnym notifyListeners(); wywoływany jest callback i można się na to zapiąć sprawdzając jaki jest dany stan, tj, najpierw pokaże loading, potem przyjdą dane to 2x wywoła się notifyListeners() - raz, żeby pokazać loading, raz żeby bindować dane. I wtedy można sprawdzić po kolei jaki stan był dla danego notifiera przy każdym wywołaniu notifyListeners(). Ogólnie bardzo spoko działa, ale trochę bieda, że provider jako pakiet nic fajnego nie udostępnia, ani też jakoś w community chyba za bardzo nie testują :P